core/src/main/scala/tech/beshu/ror/api/TestSettingsApi.scala

```diff
@@ -0,0 +1,260 @@
+    rorInstance
+      .invalidateImpersonationEngine()
+      .map { _ =>
+        TestSettingsResponse.InvalidateTestSettings.SuccessResponse("ROR Test settings are invalidated")
+      }
+  }
+
+  private def rorConfigAndTtlFrom(payload: String): EitherT[Task, UpdateTestSettings.FailedResponse, (RawRorConfig, FiniteDuration)] = {
+    for {
+      json <- EitherT.fromEither[Task](io.circe.parser.parse(payload).left.map(_ => TestSettingsResponse.UpdateTestSettings.FailedResponse("JSON body malformed")))
+      ttl <- EitherT.fromEither[Task](ttlValue(json))
+      rorConfig <- settingsValue(json)
+    } yield (rorConfig, ttl)
+  }
+
+  private def settingsValue(json: io.circe.Json) = {
+    def liftFailure(failureMessage: String) = EitherT.leftT[Task, RawRorConfig](TestSettingsResponse.UpdateTestSettings.FailedResponse(failureMessage))
+
+    json \\ "settings" match {
+      case Nil =>
+        liftFailure("Malformed settings payload - no settings key")
+      case configJsonValue :: Nil =>
+        configJsonValue.asString match {
+          case Some(configString) =>
+            EitherT(RawRorConfig.fromString(configString).map(_.left.map(error => TestSettingsResponse.UpdateTestSettings.FailedResponse(error.show))))
+          case None =>
+            liftFailure("Malformed settings payload - settings key value is not string")
+        }
+      case _ =>
+        liftFailure("Malformed settings payload - only one settings value allowed")
+    }
+  }
+
+  private def ttlValue(json: io.circe.Json) = {
+    def liftFailure(failureMessage: String) = Left(TestSettingsResponse.UpdateTestSettings.FailedResponse(failureMessage))
+
+    json \\ "ttl" match {
+      case Nil =>
+        liftFailure("Malformed ttl payload - no ttl key")
+      case ttlJsonValue :: Nil =>
+        ttlJsonValue.asString match {
+          case Some(ttlString) =>
+            parseDuration(ttlString).left.map(error => TestSettingsResponse.UpdateTestSettings.FailedResponse(error))
+          case None =>
+            liftFailure("Malformed ttl payload - ttl key value is not string")
+        }
+      case _ =>
+        liftFailure("Malformed ttl payload - only one ttl value allowed")
+    }
+  }
+
+  private def parseDuration(value: String): Either[String, FiniteDuration] = {
+    Try(Duration(value)).toOption match {
+      case Some(v: FiniteDuration) if v.toMillis > 0 => Right(v)
+      case Some(_) | None =>
+        Left(s"Cannot parse '$value' as duration.")
+    }
+  }
+
+  private def loadCurrentTestSettings()
+                                     (implicit requestId: RequestId): Task[TestSettingsResponse] = {
+    rorInstance
+      .currentTestSettings()
+      .map {
+        case TestSettings.NotConfigured =>
+          TestSettingsResponse.ProvideTestSettings.TestSettingsNotConfigured("ROR Test settings are not configured")
+        case TestSettings.Present(config, _, validTo) =>
+          val ttl = getTtl(validTo)
```

Comment:
this is the TTL that was requested. We should not recalculate the current value of TTL. The `validTo` field will tell the caller that. Maybe the description in the API doc should be more precise (if so, let's fix it)